### PR TITLE
feat:  mTLS-over-TCP buildkit

### PR DESCRIPTION
**Pull request by RealHarshThakur**
Adds flags to enable Copa to connect to a remote Buildkit listening on TCP address secured via mTLS. 
This is very similar to creating a remote buildx driver as described [here](https://docs.docker.com/build/drivers/remote/#example-remote-buildkit-in-docker-container). 

Closes https://github.com/project-copacetic/copacetic/issues/285

### CODE REVIEW

File: pkg/buildkit/drivers.go [hunk 1]
```diff
@@ -27,17 +27,40 @@ var (
...
+	client, err := client.New(ctx, bkOpts.Addr, opts...)
 	if err != nil {
 		return nil, err
 	}
 	return client, nil
 }
 
+func getCredentialOptions(bkOpts Opts) []client.ClientOpt {
+	opts := []client.ClientOpt{}
+	// addr := bkOpts.Addr
```
Comment (sozercan) [inline pkg/buildkit/drivers.go:44]: looks like leftover comment?

Comment (RealHarshThakur) [reply pkg/buildkit/drivers.go]: sorry, missed it. Removed it now :)

**Comment by RealHarshThakur** [comment]
Guessing the test failures are unrelated to this change? As I see them here as well: https://github.com/project-copacetic/copacetic/pull/287 

Please do let me know if you folks think there is a test you'd like to see. I've followed the same pattern that buildctl CLI has to set up the client.

**Comment by salaxander** [comment]
> Guessing the test failures are unrelated to this change? As I see them here as well: #287
> 
> Please do let me know if you folks think there is a test you'd like to see. I've followed the same pattern that buildctl CLI has to set up the client.

Yes it's happening across all PRs at the moment, working on addressing it

**Comment by salaxander** [comment]
This is looking good to me - @sozercan do you have a chance to review?

File: pkg/patch/cmd.go [hunk 2]
```diff
@@ -21,15 +21,18 @@ import (
...
-	buildkitAddr  string
-	timeout       time.Duration
-	ignoreError   bool
-	format        string
-	output        string
+	appImage               string
+	reportFile             string
+	patchedTag             string
+	workingFolder          string
+	buildkitAddr           string
```
Comment (cpuguy83) [inline pkg/patch/cmd.go:28]: Instead of explicitly defining these addr/cert args here, can use `buildkit.Opts` and set the flags to use `&bkopts.<Field>`

Comment (RealHarshThakur) [reply pkg/patch/cmd.go]: done :D

**Review by cpuguy83 — APPROVED** [review]

**Review by sozercan — APPROVED** [review]
Thank you! LGTM

**Pull request merged**